substrate/frame/utility/src/benchmarking.rs

```diff
@@ -92,6 +92,18 @@ mod benchmark {
 		assert_last_event::<T>(Event::BatchCompleted.into());
 	}
 
+	#[benchmark]
+	fn if_else() {
+		let main_call = Box::new(frame_system::Call::remark { remark: vec![0] }.into());
+		let fallback_call = Box::new(frame_system::Call::remark { remark: vec![1] }.into());
+		let caller = whitelisted_caller();
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(caller), main_call, fallback_call);
```

Comment:
The err branch should be slightly more costly, but difference should be negligible so it is ok.

Reply:
So adjusting with a failed main call will give the proper weights as it enters the else branch? will update this.

Thanks for the review

Reply:
You can use a root call, it will be failing, like `set_heap_page`.
This would benchmark for the failing case, which is more costly

Reply:
```rust
...
        #[benchmark]
	fn if_else() {
		let main_call = Box::new(frame_system::Call::set_heap_pages { pages: 0u64 }.into());
		let fallback_call = Box::new(frame_system::Call::remark { remark: vec![1] }.into());
		let caller = whitelisted_caller();

		#[extrinsic_call]
		_(RawOrigin::Signed(caller), main_call, fallback_call);

		assert_last_event::<T>(Event::IfElseFallbackCalled { main_error: sp_runtime::DispatchError::Module(sp_runtime::ModuleError { index: 0, error: [5, 0, 0, 0], message: None }) }.into());
	}

```

Reply:
@gui1117 this passes for `cargo test -p pallet-utility --features runtime-benchmarks`

but fails when generating weights for runtimes

Reply:
surprising :-/

Reply:
can you give an example of fail when building runtime? it seems to work for me? do you have a command to reproduce?

Reply:
Yeah, best to use other pallets for this benchmark case.

Reply:
I don't understand what you mean, can you say again?